### PR TITLE
feat: Add tests for internal/metric

### PR DESCRIPTION
**Pull request by bradbeam**
Decent coverage for basic functionality; skipping datadog for now since we dont have a test endpoint.

Signed-off-by: Brad Beam <brad@carbonrelay.com>

### CODE REVIEW

File: internal/metric/metric_test.go [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+			metric: &redskyv1alpha1.Metric{
+				Name:  "testMetric",
+				Query: "scalar(prometheus_build_info)",
+				Type:  redskyv1alpha1.MetricPrometheus,
+			},
+			obj: &corev1.ServiceList{
+				Items: []corev1.Service{
+					{
+						Spec: corev1.ServiceSpec{
+							ClusterIP: "demo.robustperception.io",
```
Comment (bradbeam) [inline internal/metric/metric_test.go:90]: saw this used in `prometheus/client_go` tests, so figure its good enough for now

**Review by andrewhare — APPROVED** [review]
👍

**Comment by andrewhare** [comment]
How much % coverage does this give for the package?

**Comment by bradbeam** [comment]
```
coverage: 49.7% of statements

datadog: 0%
jsonpath: 60.5%
metric: 73.7%
prometheus: 50%
```
Catches most of the common success paths.  Still some room to improve around the error paths.

**Pull request merged**